Peer/mainwindow.cpp

```diff
@@ -5,6 +5,10 @@
 MainWindow::MainWindow(QWidget* parent)
     : QMainWindow(parent), peer_(nullptr), ui_(new Ui::MainWindow) {
   ui_->setupUi(this);
+  logger_ = ClientLogger::Instance();
+
+  connect(logger_, SIGNAL(DisplayLog(const char*, QString)), this,
```

Comment:
We don't need signal and slots for logging something. If you need the logger, you get an instance of the logger.

Reply:
I'm using this signal only for display log in ui(will be removed later)